lnwallet/btcwallet/btcwallet.go

```diff
@@ -1373,24 +1373,256 @@ func unminedTransactionsToDetail(
+	for _, o := range outputs {
+		isOurAddress[int(o.Index)] = true
+	}
+
+	previousOutpoints := getPreviousOutpoints(&dbTx.MsgTx, inputs)
+
+	var outputDetails []lnwallet.OutputDetail
+	for i, txOut := range dbTx.MsgTx.TxOut {
+		var addresses []btcutil.Address
+		sc, outAddresses, _, err := txscript.ExtractPkScriptAddrs(
+			txOut.PkScript, chainParams,
+		)
+		if err == nil {
+			// Add supported addresses.
+			addresses = outAddresses
+		}
+
+		outputDetails = append(outputDetails, lnwallet.OutputDetail{
+			OutputType:   sc,
+			Addresses:    addresses,
+			PkScript:     txOut.PkScript,
+			OutputIndex:  i,
+			Value:        btcutil.Amount(txOut.Value),
+			IsOurAddress: isOurAddress[i],
+		})
+	}
+
+	var fee btcutil.Amount
+	if len(dbTx.Debits) == len(dbTx.MsgTx.TxIn) {
+		for _, deb := range dbTx.Debits {
+			fee += deb.Amount
+		}
+		for _, txOut := range dbTx.MsgTx.TxOut {
+			fee -= btcutil.Amount(txOut.Value)
+		}
+	}
+
+	serializedTx := dbTx.SerializedTx
+	if serializedTx == nil {
+		var buf bytes.Buffer
+		err := dbTx.MsgTx.Serialize(&buf)
+		if err != nil {
+			return nil, err
+		}
+		serializedTx = buf.Bytes()
+	}
+
+	var balanceDelta btcutil.Amount
+	for _, input := range inputs {
+		balanceDelta -= input.PreviousAmount
+	}
+	for _, output := range outputs {
+		balanceDelta += btcutil.Amount(dbTx.MsgTx.TxOut[output.Index].
+			Value)
+	}
+
+	txDetail := &lnwallet.TransactionDetail{
+		Hash:              dbTx.Hash,
+		Value:             balanceDelta,
+		BlockHash:         &dbTx.Block.Hash,
+		BlockHeight:       dbTx.Block.Height,
+		Timestamp:         dbTx.Block.Time.Unix(),
+		TotalFees:         int64(fee),
+		OutputDetails:     outputDetails,
+		RawTx:             serializedTx,
+		Label:             dbTx.Label,
+		PreviousOutpoints: previousOutpoints,
+	}
+
+	// Check if the transaction is mined.
+	if dbTx.Block.Height > 0 {
+		txDetail.NumConfirmations = currentHeight -
+			dbTx.Block.Height + 1
+	}
+
+	return txDetail, err
+}
+
 // ListTransactionDetails returns a list of all transactions which are relevant
 // to the wallet over [startHeight;endHeight]. If start height is greater than
 // end height, the transactions will be retrieved in reverse order. To include
 // unconfirmed transactions, endHeight should be set to the special value -1.
 // This will return transactions from the tip of the chain until the start
 // height (inclusive) and unconfirmed transactions. The account parameter serves
 // as a filter to retrieve the transactions relevant to a specific account. When
-// empty, transactions of all wallet accounts are returned.
+// empty, transactions of all wallet accounts are returned. Transaction hashes
+// can also be specified to fetch transactions.
 //
 // This is a part of the WalletController interface.
 func (b *BtcWallet) ListTransactionDetails(startHeight, endHeight int32,
-	accountFilter string) ([]*lnwallet.TransactionDetail, error) {
+	txHashes []*chainhash.Hash, accountFilter string,
```

Comment:
To make this non-breaking, we could add a `filterFn func(tx *lnwallet.TransactionDetail) bool` parameter that allows the caller to do filtering on the returned result. That way we don't need any additional code in here except for a loop at the end of this function that just filters the full `txDetails` slice before returning it.

Reply:
Good idea.